src/fsstorage.cc

```diff
@@ -165,33 +165,21 @@ bool FSStorage::loadMetadata(Uptane::MetaPack* metadata) {
     return false;
 
   Json::Value json = Utils::parseJSONFile(director_path / "root.json");
-  // compatibility with old clients, which store the whole metadata, not just the signed part
-  if (json.isMember("signed") && json.isMember("signatures")) json = json["signed"];
   metadata->director_root = Uptane::Root("director", json);
 
   json = Utils::parseJSONFile(director_path / "targets.json");
-  // compatibility with old clients, which store the whole metadata, not just the signed part
-  if (json.isMember("signed") && json.isMember("signatures")) json = json["signed"];
   metadata->director_targets = Uptane::Targets(json);
 
   json = Utils::parseJSONFile(image_path / "root.json");
-  // compatibility with old clients, which store the whole metadata, not just the signed part
-  if (json.isMember("signed") && json.isMember("signatures")) json = json["signed"];
   metadata->image_root = Uptane::Root("image", json);
 
   json = Utils::parseJSONFile(image_path / "targets.json");
-  // compatibility with old clients, which store the whole metadata, not just the signed part
-  if (json.isMember("signed") && json.isMember("signatures")) json = json["signed"];
   metadata->image_targets = Uptane::Targets(json);
 
   json = Utils::parseJSONFile(image_path / "timestamp.json");
-  // compatibility with old clients, which store the whole metadata, not just the signed part
-  if (json.isMember("signed") && json.isMember("signatures")) json = json["signed"];
   metadata->image_timestamp = Uptane::TimestampMeta(json);
 
   json = Utils::parseJSONFile(image_path / "snapshot.json");
-  // compatibility with old clients, which store the whole metadata, not just the signed part
-  if (json.isMember("signed") && json.isMember("signatures")) json = json["signed"];
```

Comment:
I'm worried about this.  Are we changing the on-disk format here?

Reply:
Yes. I will change the metadata load, to work with old format